bin/upload_sarif_to_gh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -eu
+
+sarif_file=$1
+owner=$2
+repo=$3
+
+if [ -z "$sarif_file" ] || [ -z "$owner" ] || [ -z "$repo" ]; then
+  echo "No argument provided. Usage: ./upload_sarif_to_gh.sh <path to .sarif report> <gh owner> <gh repo>"
+  exit 1
+fi
+
+gzip -c "$sarif_file" | base64 >sarif_base64.tmp
+
+if [[ -n $BUILDKITE_PULL_REQUEST ]]; then
+  json=$(jq -n \
+    --arg commit_sha "$BUILDKITE_COMMIT" \
+    --arg pr_number "$BUILDKITE_PULL_REQUEST" \
+    --rawfile sarif sarif_base64.tmp \
+    '{
+     "commit_sha": $commit_sha,
+     "ref": ("refs/pull/"+$pr_number+"/head"),
+     "sarif": $sarif
+   }')
+elif [[ "$BUILDKITE_BRANCH" == "$BUILDKITE_PIPELINE_DEFAULT_BRANCH" ]]; then
+  json=$(jq -n \
+    --arg commit_sha "$BUILDKITE_COMMIT" \
+    --arg branch "$BUILDKITE_BRANCH" \
+    --rawfile sarif sarif_base64.tmp \
+    '{
+     "commit_sha": $commit_sha,
+     "ref": ("refs/heads/$branch"),
+     "sarif": $sarif
+   }')
+fi
+
+echo "$json" > payload.json
```

Comment:
💭  It could probably be worth writing this `payload.json` file in a temp directory rather than the current directory?

One could argue that it doesn't really matter because the CI will clean up the checkout at the end of the job anyway. But I'm thinking of edge cases like if the step calling that command ends up having `artifact_paths: **/*.json` set up in the pipeline for other reasons (e.g with the goal of collecting unrelated `json` files) and would end up also picking up that extra `payload.json`. Granted, this is unlikely, as I'd expect us to be more precise than a generic `**/*.json` for our artifacts paths (e.g. `builds/results/*.json`), but still, just out of principle

Another option would be to give this file a more explicit name (e.g. `sarif-file-upload-payload.json` or something), so that at least if it's picked up in the artifacts by the job calling this it will be clearer what it's supposed to be.